### DISCUSSION TITLE
PlantUML support

### DISCUSSION THREAD

**Discussion by 0xDEADBEEFY** [outline/outline#3057] [Ideas]
Hello,

It would be wonderful if Outline could support PlantUML. There is some discussion about it in this issue:

https://github.com/outline/outline/issues/820

but eventually the support went the way of Draw.io instead and PlantUML seems to have been forgotten.

**Comment by ksmarty** [comment]
For computer science, this would be incredibly useful!

**Comment by pgeiem** [comment]
I am discovering outline while looking for a replacement to wiki.js and it seems a really good alternative to me. Thus PlantUML support seems  to be missing. I see there is mermaid support so it should be doable to add PlantUML too. Is there any plan for this ?

**Comment by Subtielbeek** [comment]
I would love this aswel!

**Comment by ghost** [comment]
+1

**Comment by turbcool** [comment]
The support for PlantUML would be awesome as those diagrams today could easily be generated with AI. I am really looking into documenting infrastructure with plantuml in outline.

**Comment by tonyliving** [comment]
+1

**Comment by mmua** [comment]
+1

**Comment by tmulle** [comment]
+1 Also coming from a Wiki.js self install and looking to move over to Outline instead.

**Comment by patroldo** [comment]
Hello. Here is rough example to add it. https://github.com/outline/outline/pull/10379

You can test with this link
https://editor.plantuml.com/uml/SoWkIImgAStDuNBCoKnELT2rKt3AJx9IS2mjoKZDAybCJYp9pCzJ24ejB4qjBk42oYde0jM05MDHLLoGdrUSoeLkM5u-K5sHGY9sGo6ARNHr2QY66kwGcfS2SZ00

So far functinality shall work, but need to change some icons

added a button to edit, which lead to plantuml edit-mode of diagram. But not sure i'll be able to implement that once it's edited - to propagate change back to outline. In other words - once you changed it in plantuml, you need to manually replace a link in outline

**Reply by patroldo** [reply]
Updated pull request to add plant uml icon. At that point i think it's ready

**Comment by dornstetter** [comment]
Would it be possible to set the server URI from e.g. an env var to use a self hosted plantuml instance?  Thanks!

**Reply by patroldo** [reply]
As for now - not possible. I can propose a contribution to make it available. But i can't promise that owner will accept this pull request

**Comment by stereoshoots** [comment]
+1

**Comment by nashiki** [comment]
+1

**Comment by ssamdaq** [comment]
+1
